v2/tools/generator/internal/codegen/pipeline/prune_resources_with_lifecycle_owned_by_parent.go

```diff
@@ -69,19 +75,65 @@ func PruneResourcesWithLifecycleOwnedByParent(configuration *config.Configuratio
 	return stage
 }
 
+func checkPrunedEmptyProperties(defs astmodel.TypeDefinitionSet, emptyPrunedProps []astmodel.TypeName) (astmodel.TypeDefinitionSet, error) {
+	emptyObjectVisitor := astmodel.TypeVisitorBuilder{
+		VisitObjectType: tagEmptyObjectARMProperty,
+	}.Build()
+
+	for _, emptyPrunedProp := range emptyPrunedProps {
+		// we need to add the noConversion tag on ARM type for the empty pruned property to relax the validation for convertToARM function.
+		armDef, err := GetARMTypeDefinition(defs, emptyPrunedProp)
+		if err != nil {
+			return nil, err
+		}
+
+		tempResult := make(astmodel.TypeDefinitionSet)
+		for _, def := range defs {
+			updatedDef, err := emptyObjectVisitor.VisitDefinition(def, armDef.Name())
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to visit definition %s", def.Name())
+			}
+			tempResult.Add(updatedDef)
+		}
+		defs = tempResult
+	}
+	return defs, nil
+}
+
 type misbehavingEmbeddedTypePruner struct {
-	configuration *config.Configuration
+	configuration         *config.Configuration
+	emptyPrunedProperties []astmodel.TypeName
 }
 
-func newMisbehavingEmbeddedTypeVisitor(configuration *config.Configuration) astmodel.TypeVisitor {
+func newMisbehavingEmbeddedTypeVisitor(configuration *config.Configuration) (astmodel.TypeVisitor, *misbehavingEmbeddedTypePruner) {
 	pruner := &misbehavingEmbeddedTypePruner{
 		configuration: configuration,
 	}
 
 	visitor := astmodel.TypeVisitorBuilder{
 		VisitObjectType: pruner.pruneMisbehavingEmbeddedResourceProperties,
 	}
-	return visitor.Build()
+	return visitor.Build(), pruner
+}
+
+// tagEmptyObjectARMProperty finds the empty properties in an Object and adds the ConversionTag:NoARMConversionValue property tag.
+func tagEmptyObjectARMProperty(this *astmodel.TypeVisitor, it *astmodel.ObjectType, ctx interface{}) (astmodel.Type, error) {
+	typeName := ctx.(astmodel.TypeName)
+
+	prop, ok := it.Properties().Find(func(prop *astmodel.PropertyDefinition) bool {
+		name, ok := astmodel.ExtractTypeName(prop.PropertyType())
+		if !ok {
+			return false
+		}
+		return name.Name() == typeName.Name()
+	})
+
+	if ok {
+		prop = prop.WithTag(armconversion.ConversionTag, armconversion.NoARMConversionValue)
```

Comment:
Do these tags show up in the generated code?

Reply:
Yep.

Reply:
It makes sense to have them?

Reply:
The annotation is only consumed within the generator, it has no runtime function - so I think it would be ideal to clean up so they don't show in the generated code. Not sure on the ROI of the effort though.

Reply:
I think we have another tag that's only on the ARM type which we also have left in the code for now.

I see the ARM types as pretty internal so I'm OK if it's just on the ARM type. If it's going to be on the k8s type too we should remove it.

My 2c is one of these options makes sense:
1. Just leave it.
2. Leave it but file an issue to clean it (and any other tags) up.

I think I tend towards option 1 but I'm not married to it.